Add ISPC language support

aneshlya opened the pull request:
Adds the ISPC language, full name "Intel® Implicit SPMD Program Compiler" https://github.com/ispc/ispc, extension `.ispc`, along with its header files `.isph`.
ISPC is a C-based language with extensions for SIMD programming.
ISPC is used across different game engines (e.g. Unreal Engine, Unity), production renderers (e.g Dreamworks' MoonRay) and ray tracing libraries (e.g. Intel(R) Ospray)

Closes https://github.com/github-linguist/linguist/issues/5042

- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
      -  https://github.com/search?type=code&q=NOT+is%3Afork+path%3A*.ispc - 8.7k
      -  ~~https://github.com/search?type=code&q=NOT+is%3Afork+path%3A*.isph - 644~~
  - [x] I have included a real-world usage sample for all extensions added in this PR:
         - [samples/ISPC/VideoFunctions.ispc](https://github.com/xLightsSequencer/xLights/blob/master/xLights/effects/ispc/VideoFunctions.ispc), [GPL-3](https://github.com/xLightsSequencer/xLights?tab=GPL-3.0-1-ov-file)
         - [samples/ISPC/volume.ispc](https://github.com/ispc/ispc/blob/main/examples/cpu/volume_rendering/volume.ispc), [BSD3C](https://github.com/ispc/ispc?tab=BSD-3-Clause-1-ov-file)
         - [samples/ISPC/program_city.ispc](https://github.com/re-esper/CPUShadertoy/blob/master/program_city.ispc), [MIT](https://github.com/re-esper/CPUShadertoy?tab=MIT-1-ov-file)

  - [x] I have included a syntax highlighting grammar: [[ispc.syntax]](https://github.com/ispc/ispc.syntax)
  - [x] I have added a color
    - Hex value: `#2D68B1`
    - Rationale: it is used in a header on official website: https://ispc.github.io/ispc.html
  - [ ] I have updated the heuristics to distinguish my language from others using the same extension. N/A. There are no languages with the same extension.

lildude commented:
> - https://github.com/search?type=code&q=NOT+is%3Afork+path%3A*.isph - 644

This low usage will block this PR. You might want to consider removing it. People can use an override for these files until usage is sufficient to add support.

aneshlya commented:
> This low usage will block this PR. You might want to consider removing it. People can use an override for these files until usage is sufficient to add support.

I removed `.isph` extension and added more `.ispc` samples.

lildude reviewed (approved):
LGTM. Thanks.

> [!IMPORTANT]
> The changes in this PR will not appear on GitHub until the next release has been made and deployed. See [here](https://github.com/github/linguist/blob/master/docs/troubleshooting.md#my-linguist-pr-has-been-merged-but-github-doesnt-reflect-my-changes) for more details.
